server/tserver/src/main/java/org/apache/accumulo/tserver/metrics/TabletServerScanMetrics.java

```diff
@@ -44,8 +47,18 @@ public void addYield(long value) {
     yields.record(value);
   }
 
+  public void incrementOpenFiles(int numOpenFiles) {
+    openFiles.addAndGet(numOpenFiles);
+  }
+
+  public void decrementOpenFiles(int numOpenFiles) {
```

Comment:
In the cases where this is used, will the value of `openFiles` ever actually be decremented? It looks like `incrementOpenFiles()` and `decrementOpenFiles()` will both do the same thing at the moment. Is the idea that we always pass a negative number to `decrementOpenFiles()`?

Reply:
LOL - yes, in my mind, that's what was happening. I should probably multiple the argument by -1. Good catch.

Reply:
Good catch. I don't think its possible that the fileManager will return a negative number though since it is just giving the size of the ArrayList. You could put assert checks at the beginning of each method to make sure the parameter is positive, then always multiplying by -1 for the decrementing.

Reply:
Otherwise, the parameter name is odd with the way you have it now since you can't have negative "numOpenFiles". If you want to allow passing a negative parameter you could name it "deltaOpenFiles" or something similar since the parameter to `addAndGet` is called "delta". https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/atomic/AtomicInteger.html#addAndGet(int)

Reply:
I agree, I think it makes more sense to either:
keep both methods and only allowing non-negative numbers for each, multiplying by -1 for decrement()
**or,**
just use one method that will add to `openFiles` regardless of whether the input is positive or negative. (preserving the functionality of `addAndGet()`)

Reply:
See 4b0db48